Added design specification document

mukulkhanna opened the pull request:
- Added a design specification document for the project.
- It contains information about the design decisions made that can help in the future.

j-rivero commented on design_spec.md:
````diff
@@ -0,0 +1,205 @@
+- category 3
+```
+
+### 4.3 Repository structure
````
This repository section is detailing the low level implementation of how the index work. I think that there is value in these sentences but I'm not sure if we want it mixed with high level design sections above. @mukulkhanna do you think that we can create a new section (at the end of the doc) and place the repository structure in there?

mukulkhanna replied:
I think that except for information about the ```_categories``` folder, we can have the other parts of repository structure just in the README of the repository.

Or else, I can move this section to #5 (before the 'hosting' section).

j-rivero replied:
Whatever you prefer, both work for me.

j-rivero commented on design_spec.md:
```diff
@@ -0,0 +1,205 @@
+
+### 4.2 Index structure
+
+The documentation index items have been divided by categories and subcategories.
```
For me is clear that the maximum deep level for the index is two (categories >> subcategories) but I would like to see it written explicitly somewhere in these sentences.

mukulkhanna replied:
Okay, I will add that.

mukulkhanna commented:
Added in CONTRIBUTING.md -
- Info about how to preview changes on local server
- Table of contents for easier navigation to items that are lower in the document

Updated in design_spec.md - 
- AWS -> Github in hosting section
- Explicit info about category and subcategory levels
- Moved the repo structure lower in the document and removed some lower level info
- Table of contents
